File: src/main/scala/io/github/oxnz/Ingrid/tx/TxHttpExecCallback.scala
package io.github.oxnz.Ingrid.tx

import org.apache.http.concurrent.FutureCallback

class TxHttpExecCallback extends FutureCallback[TxResult] {
  override final def cancelled(): Unit = {
    // no-op
  }

  override def completed(result: TxResult): Unit = {
  }

  override def failed(ex: Exception): Unit = {
  }
}

File: src/main/java/io/github/oxnz/Ingrid/tx/TxHttpExecCallback.java
package io.github.oxnz.Ingrid.tx;

import org.apache.http.concurrent.FutureCallback;

public class TxHttpExecCallback implements FutureCallback<TxResult> {
    @Override
    public void completed(TxResult result) {

    }

    @Override
    public void failed(Exception ex) {

    }

    @Override
    public final void cancelled() {
        // no-op
    }
}
